component_sdk/python/kfp_component/google/bigquery/_query.py

```diff
@@ -56,13 +80,41 @@ def _get_job(client, job_id):
     except exceptions.NotFound:
         return None
 
+def _prepare_dataset_ref(client, dataset_id, output_gcs_path):
+    if not output_gcs_path and not dataset_id:
+        return None
+    
+    if not dataset_id:
+        dataset_id = 'kfp_tmp_dataset'
+    dataset_ref = client.dataset(dataset_id)
+    dataset = _get_dataset(client, dataset_ref)
+    if not dataset:
+        logging.info('Creating dataset {}'.format(dataset_id))
+        dataset = _create_dataset(client, dataset_ref)
+    return dataset_ref
+
+def _get_dataset(client, dataset_ref):
+    try:
+        return client.get_dataset(dataset_ref)
+    except exceptions.NotFound:
+        return None
+
+def _create_dataset(client, dataset_ref):
+    dataset = bigquery.Dataset(dataset_ref)
+    dataset.location = "US"
```

Comment:
Should the location be configurable here?

Reply:
Done